CONTRIBUTING.md

```diff
@@ -1,7 +1,332 @@
+```
+
+and not:
+
+```
+[example](example.md#heading-text)
+```
+
+When referring literally to a Handbook section, the section title should be
+placed in double-quotes. Otherwise, double-quotes are not required. For example:
+
+```
+For more information, please read the "[Power Management](./power-management.md)" section.
+```
+
+and
+
+```
+Void provides facilities to assist with [power management](./power-management.md).
+```
+
+#### Man Page Links
+
+The first reference to a command or man page must be a link to the relevant man
+page on `https://man.voidlinux.org/`.
+
+The link text must contain the title section number in parenthesis, and contain
+no formatting. For example: [man(1)](https://man.voidlinux.org/man.1), not
+[`man(1)`](https://man.voidlinux.org/man.1).
+
+#### Auto Links
+
+Auto links (links with the same title as URL) should use the following notation:
+
+```
+<https://www.example.com/>
+```
+
+They should not be formatted like this:
+
+```
+[https://www.example.com/](https://www.example.com/)
+```
+
+#### Checking links
+
+If you're including new links (either internal or external) in the docs or
+changing the current file structure, you should make use of the
+`mdbook-linkcheck` package:
+
+```
+$ mdbook-linkcheck -s
+```
+
+This will verify all the references, and warn you if there are any issues. If
+any link you're using is correct but generating errors for some reason, you can
+add its domain to the exclude list in `book.toml`, under the
+`[mdbook.linkcheck]` key.
+
+`mdbook-linkcheck` is used by the void-docs repository's `check.sh` script,
+which must be [run locally before submitting a pull request](#making-changes).
+
+### Case
+
+Filenames and directories should use [kebab
+case](https://en.wikipedia.org/wiki/Kebab_case) when splitting words. For
+example the filename should be `post-install.md` not `postinstall.md`.
+
+Words that are part of trademarks or well known package names are exempt from
+this rule. Examples include `PulseAudio` and `NetworkManager` which are well
+known by their squashed names.
+
+### Voice
+
+Prefer the active imperative voice when writing documentation. Consider the
+following examples:
+
+> Now we need to install the CUPS drivers and configure them.
+
+This version is conversational and friendlier, but contains unnecessary language
+that may not be as clear to an ESL reader.
+
+> Install and configure the CUPS drivers, then configure them as shown.
+
+This version contains a clear command to act, and a follow up that shows what
+will be done next. It is clear both to native English speakers, ESL readers, and
+to translators.
+
+### Notes
+
+Notes should only be used sparingly, and for non-critical information. They
+should begin with "Note: ", and not be block-quoted with `>`. For example, the
+Markdown should look like:
+
+```
+Note: You can also use program X for this purpose.
+```
+
+and not:
+
+```
+> You can also use program X for this purpose.
+```
```

Comment:
Mention warnings around here? 

And mention that in most cases the text flows better if using a construct like "It should be noted that..." ?

Reply:
Well, i'm not sure #324 was clearly resolved? (And generating the PDF is the sort of situation i anticipated in that discussion, based on my experiences with other projects - the code will be made a bit more complex by having to distinguish warning-bolding from non-warning bolding.) What's your take on that?

Re. notes, so you think we should change this part to just say something like "Notes should be indicated with the phrase "It should be noted that ..."?

Reply:
I left a comment there. Most of the discussion is a bit beyond me, though.

> Re. notes, so you think we should change this part to just say something like "Notes should be indicated with the phrase "It should be noted that ..."?

I prefer this style, having a `Note:` feels like it breaks the flow a bit. What do you think?

Reply:
> I left a comment there. Most of the discussion is a bit beyond me, though.

Okay. i'd like to know @bobertlo's preferences on this.

> I prefer this style, having a Note: feels like it breaks the flow a bit. What do you think?

i agree. The more i encounter documentation with visual callout of notes, the more i dislike it. @bobertlo, thoughts?

Reply:
i've just changed the style guide's subsection on notes appropriately.

Reply:
Great. What did you think of my usage of `**Warning**`: in the ARM guide?

Reply:
That style is probably still my overall preference, so, works for me. :-)